extensions/libarchive/MergeContent.cpp

```diff
@@ -213,18 +213,19 @@ bool MergeContent::processBin(core::ProcessContext *context, core::ProcessSessio
         });
   }
 
-  std::shared_ptr<core::FlowFile> merge_flow = std::static_pointer_cast<FlowFileRecord>(session->create());
+  std::shared_ptr<core::FlowFile> merge_flow = std::static_pointer_cast<FlowFileRecord>(session.create());
```

Comment:
I think it might be better to introduce a function to processSession that checks if the flowfile is in a valid transferred state, and then let a RAII take care of the removal instead. This way we dont have to keep in mind that we have to remove it during the various early return paths.
something like this should do it
```
  auto remove_merge_flow = gsl::finally([&](){
    if (!session.hasBeenTransferred(*merge_flow))
      session.remove(merge_flow);
  });
```
whre the hasbeentransferred should check if the updated_relationships_ or added_flowfiles_ maps contain a nonnull relationship* for the flowfile

Reply:
Good idea, updated in 8a269048b9b6f451f0baedd584a4358286908a60